parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -268,10 +269,11 @@ sqliteCodebase debugName root localOrRemote action = do
           Sqlite.runTransaction conn CodebaseOps.getRootBranchExists
 
         putRootBranch :: TVar (Maybe (Sqlite.DataVersion, Branch Sqlite.Transaction)) -> Branch m -> m ()
-        putRootBranch rootBranchCache branch1 =
-          withRunInIO \runInIO ->
+        putRootBranch rootBranchCache branch1 = do
+          withRunInIO \runInIO -> do
             Sqlite.runTransaction conn do
               CodebaseOps.putRootBranch rootBranchCache (Branch.transform (Sqlite.unsafeIO . runInIO) branch1)
+            liftIO $ Sqlite.unsafeUnTransaction (CodebaseOps.saveRootNamesIndex (Branch.toNames $ Branch.head branch1)) conn
```

Comment:
Is there a reason we don't do this in the same transaction?

Reply:
Haha you must've caught this just a minute before I removed it entirely 😋